ext/gd/gd.c

```diff
@@ -3082,6 +3082,12 @@ static void php_imagettftext_common(INTERNAL_FUNCTION_PARAMETERS, int mode)
 		im = php_gd_libgdimageptr_from_zval_p(IM);
 	}
 
+	// FT_F26Dot6 is a signed long alias
+	if (ptsize < (double)LONG_MIN / 64 || ptsize > (double)LONG_MAX / 64) {
```

Comment:
`(double)LONG_MIN / 64` is not the same as `LONG_MIN / 64`. The former equals 144115188075855871 and the latter equals 144115188075855872.0 (on 64-bit). Therefore, there is a mismatch between the condition and the error message.